docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md

```diff
@@ -4,17 +4,677 @@ title: Connector SDK
+controlled in a secure location and referenced in a connector's properties using a placeholder
+pattern `secrets.*`. To make this mechanism as robust as possible, secret handling comes with
+the connector SDK out of the box. That way, all connectors can use the same standard way of
+handling secrets in input data.
+
+The SDK allows replacing secrets in input data as late as possible to avoid passing them around
+in the environments that handle connector invocation. Therefore, we do not pass secrets into the
+connector function in clear text but only as placeholders that you can replace from
+within the connector function. To initiate the secret replacement from the connector function,
+use the `ConnectorContext` object's `replaceSecrets` method as shown in the
+[runtime logic](#runtime-logic) section:
+
+```java
+...
+  @Override
+  public Object execute(ConnectorContext context) throws Exception {
+    ...
+    // (3)
+    context.replaceSecrets(connectorRequest);
+    ...
+  }
+...
+```
+
+This will instruct the context to search for and replace secret placeholders in the object passed to it.
+The secret store that is present in the connector runtime environment that invokes the connector function
+takes care of that. Every environment can define its own way of providing such a secret store. Consult
+the [runtime environments](#runtime-environments) section to learn more about them.
+
+To replace secrets for the `connectorRequest` using the API, the object needs to implement the
+`ConnectorInput` interface. Using this interface's `replaceSecrets` method, you can use the provided
+`SecretStore` to search for and replace secrets in the properties that should support this in your
+connector. Furthermore, the `ConnectorInput` interface comes with convenience methods to allow handling
+nested objects as well with `replaceSecretsIfNotNull`:
+
+```java
+package io.camunda.connector;
+
+import io.camunda.connector.api.ConnectorInput;
+import io.camunda.connector.api.SecretStore;
+
+public class MyConnectorRequest implements ConnectorInput {
+
+  private String message;
+  private Authentication authentication;
+
+  @Override
+  public void replaceSecrets(final SecretStore secretStore) {
+    replaceSecretsIfNotNull(authentication, secretStore);
+  }
+}
+```
+
+```java
+package io.camunda.connector;
+
+import io.camunda.connector.api.ConnectorInput;
+import io.camunda.connector.api.SecretStore;
+import java.util.Objects;
+
+public class Authentication implements ConnectorInput {
+
+  private String user;
+  private String token;
+
+  @Override
+  public void replaceSecrets(final SecretStore secretStore) {
+    token = secretStore.replaceSecret(token);
+  }
+}
+```
+
+Using this approach, you can replace secrets in your whole input data structure with one
+initial call from the central connector function.
+
+## Testing
+
+Making sure your connector's business logic works as expected is vital to developing the connector.
```

Comment:
Minor tweak:
Update text from "Making sure your connector's..."
to "Ensuring your connector's..."

Reply:
Updated